pkg/lib/serverTime.js

```diff
@@ -258,42 +268,30 @@
         return status;
     };
 
-    const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
-                                                { superuser: "try" });
-
-    self.get_custom_ntp = function () {
-        /* We only support editing the configuration of
-         * systemd-timesyncd, by dropping a file into
-         * /etc/systemd/timesyncd.conf.d.  We assume that timesyncd is
-         * used when:
-         *
-         * - systemd-timedated is answering for
-         *   org.freedesktop.timedate1 as opposed to, say, timedatex.
-         *
-         * - systemd-timesyncd is enabled (false if chrony is being used)
-         *
-         * The better alternative would be to have an API in
-         * o.fd.timedate1 for managing the list of NTP server
-         * candidates.
-         */
+    function get_timesync_backend() {
+        // We assume chronyd > ntpd > timesyncd.  If the assumption
+        // turns out to be wrong, we will tell the user afterwards.
+        if (chronyd_service.exists)
+            return "chronyd";
+        else if (ntpd_service.exists)
+            return null;
+        else if (timesyncd_service.exists)
+            return "timesyncd";
+        else
+            return null;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230118-144535-6ccd176d-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)